src/main/java/org/opentripplanner/middleware/controllers/api/OtpUserController.java

```diff
@@ -1,19 +1,111 @@
 
 /**
- * Implementation of the {@link AbstractUserController} for {@link OtpUser}.
+ * Implementation of the {@link AbstractUserController} for {@link OtpUser}. This controller also contains
+ * OtpUser-specific endpoints for verifying the user's phone number by SMS.
  */
 public class OtpUserController extends AbstractUserController<OtpUser> {
+    private static final Logger LOG = LoggerFactory.getLogger(OtpUserController.class);
+
+    private static final String CODE_PARAM = "code";
+    protected static final String CODE_PATH = "/:" + CODE_PARAM;
+    private static final String VERIFY_ROUTE = "/verify_sms";
+
     public OtpUserController(String apiPrefix) {
         super(apiPrefix, Persistence.otpUsers, "secure/user");
     }
 
+    @Override
+    protected void buildEndpoint(ApiEndpoint baseEndpoint) {
+        LOG.info("Registering path {}.", ROOT_ROUTE + "verify");
+
+        // Add the api key route BEFORE the regular CRUD methods
+        ApiEndpoint modifiedEndpoint = baseEndpoint
+            .get(path(ROOT_ROUTE + ID_PATH + VERIFY_ROUTE)
+                    .withDescription("Request an SMS verification to be sent to an OtpUser's phone number.")
+                    .withPathParam().withName(ID_PARAM).withDescription("The id of the OtpUser.").and()
+                    .withResponseType(VerificationResult.class),
+                this::sendVerificationText, JsonUtils::toJson
+            )
+            .post(path(ROOT_ROUTE + ID_PATH + VERIFY_ROUTE + CODE_PATH)
+                    .withDescription("Verify an OtpUser's phone number with a verification code.")
+                    .withPathParam().withName(ID_PARAM).withDescription("The id of the OtpUser.").and()
+                    .withPathParam().withName(CODE_PARAM).withDescription("The SMS verification code.").and()
+                    .withResponseType(VerificationResult.class),
+                this::verifyPhoneWithCode, JsonUtils::toJson
+            );
+        // Add the regular CRUD methods after defining the controller-specific routes.
+        super.buildEndpoint(modifiedEndpoint);
+    }
+
     @Override
     protected OtpUser getUserProfile(Auth0UserProfile profile) {
         return profile.otpUser;
     }
+
+    /**
+     * HTTP endpoint to send an SMS text to an {@link OtpUser}'s phone number with a verification code. This is used
+     * during user signup (or if a user wishes to change their notification preferences to use a new un-verified phone
+     * number).
+     */
+    public VerificationResult sendVerificationText(Request req, Response res) {
+        OtpUser otpUser = getEntityForId(req, res);
+        if (otpUser.phoneNumber == null) {
+            logMessageAndHalt(req, HttpStatus.NOT_FOUND_404, "User must have valid phone number to verify.");
+            return null;
+        }
+        Verification verification = NotificationUtils.sendVerificationText(otpUser.phoneNumber);
```

Comment:
This flow blurs the natural order of things, as this function assumes that a phone number is already assigned in the `OtpUser` record, whereas on the UI flow, we would want to validate that number before we do assign it to the `OtpUser` record. I'm adding this comment to #70.